### PR TITLE
Update gradle to 4.5

### PR DESCRIPTION
**Pull request by BanzaiMan**
## What is the problem that this PR is trying to fix?

Update Gradle to 4.5, as 4.0 is incompatible with Java 9

See https://github.com/travis-ci/travis-ci/issues/8975.

## How can you make sure the change works as expected?

According to the GitHub issue above, one can run:

```sh-session
$ gradle -PTestLoggingFull=true check
```

### CODE REVIEW

File: cookbooks/travis_build_environment/attributes/default.rb [hunk 1]
```diff
@@ -228,7 +228,7 @@
 default['travis_build_environment']['haskell_default_ghc'] = '7.10.3'
 default['travis_build_environment']['haskell_default_cabal'] = '1.22'
 
-gradle_version = '4.0.1'
+gradle_version = '4.5'
 default['travis_build_environment']['gradle_version'] = gradle_version
 default['travis_build_environment']['gradle_url'] = "https://services.gradle.org/distributions/gradle-#{gradle_version}-bin.zip"
 default['travis_build_environment']['gradle_checksum'] = 'd717e46200d1359893f891dab047fdab98784143ac76861b53c50dbd03b44fd4'
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/attributes/default.rb]: @BanzaiMan Are you able to fill in the new SHA-256 checksum, too?

**Comment by dpolivaev** [comment]
@meatballhat  @BanzaiMan  any update on this?

File: cookbooks/travis_build_environment/attributes/default.rb [hunk 2]
```diff
@@ -228,7 +228,7 @@
 default['travis_build_environment']['haskell_default_ghc'] = '7.10.3'
 default['travis_build_environment']['haskell_default_cabal'] = '1.22'
 
-gradle_version = '4.0.1'
+gradle_version = '4.5'
```
Comment (dpolivaev) [inline cookbooks/travis_build_environment/attributes/default.rb]: Current latest version is 4.6
~~~~
gradle_version='4.6'
default['travis_build_environment']['gradle_version'] = gradle_version
default['travis_build_environment']['gradle_url'] = "https://services.gradle.org/distributions/gradle-#{gradle_version}-bin.zip"
 default['travis_build_environment']['gradle_checksum'] = '98bd5fd2b30e070517e03c51cbb32beee3e2ee1a84003a5a5d748996d4b1b915'
~~~~

Could you please update your configuration?

**Comment by Lyoness** [comment]
Current gradle version is [4.10.2](https://github.com/travis-ci/travis-cookbooks/blob/master/cookbooks/travis_build_environment/attributes/default.rb#L236) in master.  Closing out.

**Pull request closed without merging**